text/vec-set.md

```diff
@@ -55,25 +52,29 @@ conditions first, and then insert the new member only after we are sure it is sa
 an example of the mnemonic idiom, "**verify first write last**".
 
 ```rust, ignore
-pub fn add_member(origin) -> DispatchResult {
+#[pallet::weight(10_000)]
+pub fn add_member(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
 	let new_member = ensure_signed(origin)?;
 
 	let mut members = Members::<T>::get();
-	ensure!(members.len() < MAX_MEMBERS, Error::<T>::MembershipLimitReached);
-
-	// We don't want to add duplicate members, so we check whether the potential new
-	// member is already present in the list. Because the list is always ordered, we can
-	// leverage the binary search which makes this check O(log n).
-	match members.binary_search(&new_member) {
-		// If the search succeeds, the caller is already a member, so just return
-		Ok(_) => Err(Error::<T>::AlreadyMember.into()),
-		// If the search fails, the caller is not a member and we learned the index where
-		// they should be inserted
-		Err(index) => {
-			members.insert(index, new_member.clone());
-			Members::<T>::put(members);
-			Self::deposit_event(RawEvent::MemberAdded(new_member));
-			Ok(())
+	ensure!(
+		members.len() < MAX_MEMBERS,
+		Error::<T>::MembershipLimitReached
+		);
+
+		// We don't want to add duplicate members, so we check whether the potential new
+		// member is already present in the list. Because the list is always ordered, we can
+		// leverage the binary search which makes this check O(log n).
+		match members.binary_search(&new_member) {
+			// If the search succeeds, the caller is already a member, so just return
+			Ok(_) => Err(Error::<T>::AlreadyMember.into()),
+			// If the search fails, the caller is not a member and we learned the index where
+			// they should be inserted
+			Err(index) => {
+				members.insert(index, new_member.clone());
+				Members::<T>::put(members);
+				Self::deposit_event(Event::MemberAdded(new_member));
+				Ok(().into())
 		}
 	}
 }
```

Comment:
Is the formatting right here? The syntax highlighting is confusing github. Not sure why though. Maybe this is just fine.